src/stdatamodels/model_base.py

```diff
@@ -533,48 +540,40 @@ def save(self, path, dir_path=None, *args, **kwargs):
             path_head, path_tail = os.path.split(path(self.meta.filename))
         else:
             path_head, path_tail = os.path.split(path)
-        base, ext = os.path.splitext(path_tail)
+        ext = Path(path_tail).suffix
         if isinstance(ext, bytes):
             ext = ext.decode(sys.getfilesystemencoding())
 
         if dir_path:
             path_head = dir_path
-        output_path = os.path.join(path_head, path_tail)
+        output_path = os.path.join(path_head, path_tail)  # noqa: PTH118
```

Comment:
same here, can os.path instances be fixed?

Reply:
`output_path` is returned below (as a string). Rather than convert to a `Path` to do the join then back to a string to return I think it's simpler and more readble to keep it as a string.